### ISSUE TITLE
Color camera support

### ISSUE THREAD

**Issue by gmiller-whus** [mabl/PyPylon#24]
Any info on when or if support for color cameras will be available?  Any info on why it is disabled at this time?  Thanks.

**Comment by vgundecha** [comment]
Any update on this one? Did you get it working?
Thanks

**Comment by gmiller-whus** [comment]
Yes, I did get it working.  I am now able to capture an image from a color camera using the “RGB8” format.  I need more testing before I branch/merge the code.

Gasry

**Comment by AkhilAKrasH** [comment]
Hi!
have you tried your solution? Do you plan to commit your code?

Thx

**Comment by gmiller-whus** [comment]
Hi,

Yes, I have tries my solution and it works great with a color camera.  I do not have a B&W camera to confirm that it still works properly that way.

I do intent to commit my code in the next few days.

Thx.

**Comment by Wing0** [comment]
For those who run into this issue while the support for color cameras is not there yet, here's a quick fix:

Comment lines 224 and 225 on PyPylon/cython/factory.pyx
Change line 251 to: `img_data = np.frombuffer((<char*>img.GetBuffer())[:img.GetImageSize()], dtype='uint8')#+bits_per_pixel_prop[3:])`
install pypylon with `python setup.py install`

This results in the different color channels being repeated in a single one channel image. The snippet below takes the picture and returns the image in BGR colors which can be directly worked on with OpenCV.

```
    cam = pypylon.factory.create_device(devices[0])
    cam.open()
    cam.properties['PixelFormat'] = 'RGB8'
    images = cam.grab_images(1)
    img = [o for o in images][0]

    out = np.zeros((img.shape[0], img.shape[1] / 3, 3), dtype=np.uint8)
    for j in range(img.shape[1]):
        c = int(j / 3)
        out[:, c, 2 - (j % 3)] = img[:, j]
```

**Comment by Terrapon** [comment]
@Wing0 
I have an issue with your solution. the code return an error
`out = np.zeros((img.shape[0], img.shape[1] / 3, 3), dtype=np.uint8)
TypeError: 'float' object cannot be interpreted as an integer`

seems like I have to cast "img.shape[1] / 3" to int. otherwise Python stores it as a float and now it's working

As @Wing0 says in the next comment, I'm using python 3.6

**Comment by Wing0** [comment]
@Terrapon Thanks for pointing this out! You're probably using python 3.x whereas I have python 2.7.13 and division always returns an integer. Other than that, I hope it's working out for you.
